.github/workflows/bump_submodules.yml

```diff
@@ -0,0 +1,46 @@
+name: Bump Submodule
+
+on:
+  schedule:
+    - cron: '0 0 * * 0,4'  # Run at 00:00 on Sunday and Thursday
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/bump_submodules.yml"
+
+jobs:
+  bump_submodule:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Bump torch-mlir submodule
+      run: |
+        cd third_party/torch-mlir
+        git fetch origin
+        git checkout origin/main
+        torch_mlir_commit=$(git rev-parse HEAD)
+        cd ../..
+        git add third_party/torch-mlir
+        git commit -m "Bump torch-mlir submodule to latest version"
+        git push
+        echo "torch_mlir_commit=$torch_mlir_commit" >> $GITHUB_OUTPUT
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v3
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        branch: bump-torch-mlir-submodule
+        title: 'Bump torch-mlir submodule to latest version'
```

Comment:
Especially if this runs on a cron, the titles should be unique (otherwise we could have multiple PRs with the same title). Could include the commit here